### PR TITLE
[TypeScript][VA/Skill] Integrate specific cultures

### PR DESCRIPTION
**Pull request by DiegoCardozo94**
## Purpose

_What is the context of this pull request? Why is it being done?_

We updated the _Virtual Assistant_ and the _Skill_ language cultures to improve the functionality of the **multilanguage** implementation. 

## Changes
_Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)_

- Rename different folder and files with a specific _locale_ name
- Use of the specific _locale_ name for the _deployment scripts_ files
- Changes in _mainDialog_ implementation

## Tests
_Is this covered by existing tests or new ones? If no, why not?_
*-*

## Testing steps

1. Deploy a Virtual Assistant with both languages (eg. _en-us, es-mx_).
2. Verify in the bot solution that the resource folder contains the specific cultures.
3. Execute the Virtual Assistant with any of the deployed specific cultures
4. Check the behavior of the Virtual Assistant communication

## Feature Plan
_Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues._
*-*

#### General
- [ ] I have commented my code, particularly in hard-to-understand areas	
- [ ] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

#### Bots
- [X] I have replicated language model changes across the English, French, Italian, German, Spanish, and Chinese `.lu` files and validated that deployment is successful

#### Deployment Scripts
- [X] I have replicated my changes in the **Virtual Assistant Template** and **Sample** projects
- [X] I have replicated my changes in the **Skill Template** and **Sample** projects

### CODE REVIEW

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/index.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ const _pick = require(`lodash/pick`);
 const _kebabCase = require(`lodash/kebabCase`);
 const _camelCase = require(`lodash/camelCase`);
 const templateName = "sample-assistant";
-const languages = [`zh`, `de`, `en`, `fr`, `it`, `es`];
+const languages = [`zh-cn`, `de-de`, `en-us`, `fr-fr`, `it-it`, `es-mx`];
```
Comment (Batta32) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/index.js]:
```suggestion
const languages = [`zh-cn`, `de-de`, `en-us`, `fr-fr`, `it-it`, `es-es`];
```

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/index.js [hunk 2]
```diff
@@ -22,32 +22,32 @@ let finalAssistantName = "";
...
   {
     name: "Italian",
-    value: "it",
+    value: "it-it",
     checked: true
   },
   {
     name: "Spanish",
-    value: "es",
+    value: "es-mx",
```
Comment (Batta32) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/index.js]:
```suggestion
    value: "es-es",
```

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/src/copier.js [hunk 3]
```diff
@@ -6,7 +6,7 @@
 const { join } = require(`path`);
 const templateFiles = new Map();
 const deploymentTemplateFiles = new Map();
-const allLanguages = [`zh`, `de`, `en`, `fr`, `it`, `es`];
+const allLanguages = [`zh-cn`, `de-de`, `en-us`, `fr-fr`, `it-it`, `es-mx`];
```
Comment (Batta32) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/src/copier.js]:
```suggestion
const allLanguages = [`zh-cn`, `de-de`, `en-us`, `fr-fr`, `it-it`, `es-es`];
```

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/index.js (hunk 1 above)
Comment (Batta32) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/index.js]:
```suggestion
const languages = [`zh-cn`, `de-de`, `en-us`, `fr-fr`, `it-it`, `es-es`];
```

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/index.js (hunk 2 above)
Comment (Batta32) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/index.js]:
```suggestion
    value: "es-es",
```

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/src/copier.js (hunk 3 above)
Comment (Batta32) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/src/copier.js]:
```suggestion
const allLanguages = [`zh-cn`, `de-de`, `en-us`, `fr-fr`, `it-it`, `es-es`];
```

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/sample-assistant/deployment/scripts/deploy_cognitive_models.ps1 [hunk 4]
```diff
@@ -114,7 +114,7 @@ $settings = @{ defaultLocale = $languageArr[0]; cognitiveModels = New-Object PSO
 Write-Host "> Deploying cognitive models ..."
 foreach ($language in $languageArr)
 {
-    $langCode = ($language -split "-")[0]
+	$langCode = $language
```
Comment (Batta32) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/sample-assistant/deployment/scripts/deploy_cognitive_models.ps1]: Is it necessary this assignment? Can we use `$language` for the rest of the `foreach`?

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/sample-assistant/src/index.ts [hunk 5]
```diff
@@ -50,8 +50,8 @@ import { skills as skillsRaw } from './skills.json';
 // tslint:disable-next-line: no-floating-promises
 i18next.use(i18nextNodeFsBackend)
     .init({
-        fallbackLng: 'en',
-        preload: [ 'de', 'en', 'es', 'fr', 'it', 'zh' ],
+        fallbackLng: 'en-us',
+        preload: ['de-de', 'en-us', 'es-mx', 'fr-fr', 'it-it', 'zh-cn'],
```
Comment (Batta32) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/sample-assistant/src/index.ts]:
```suggestion
        preload: ['de-de', 'en-us', 'es-es', 'fr-fr', 'it-it', 'zh-cn'],
```

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/sample-assistant/test/helpers/botTestBase.js [hunk 6]
```diff
@@ -38,8 +38,8 @@ async function initConfiguration() {
     // Configure internationalization and default locale
     await i18next.use(i18nextNodeFsBackend)
     .init({
-        fallbackLng: 'en',
-        preload: [ 'de', 'en', 'es', 'fr', 'it', 'zh' ],
+        fallbackLng: 'en-us',
+        preload: ['de-de', 'en-us', 'es-mx', 'fr-fr', 'it-it', 'zh-cn'],
```
Comment (Batta32) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/sample-assistant/test/helpers/botTestBase.js]:
```suggestion
        preload: ['de-de', 'en-us', 'es-es', 'fr-fr', 'it-it', 'zh-cn'],
```

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/skill/index.js [hunk 7]
```diff
@@ -13,7 +13,7 @@ const _pick = require(`lodash/pick`);
 const _kebabCase = require(`lodash/kebabCase`);
 const _camelCase = require(`lodash/camelCase`);
 const templateName = "sample-skill";
-const languages = [`zh`, `de`, `en`, `fr`, `it`, `es`];
+const languages = [`zh-cn`, `de-de`, `en-us`, `fr-fr`, `it-it`, `es-mx`];
```
Comment (Batta32) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/skill/index.js]:
```suggestion
const languages = [`zh-cn`, `de-de`, `en-us`, `fr-fr`, `it-it`, `es-es`];
```

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/skill/index.js [hunk 8]
```diff
@@ -22,32 +22,32 @@ let finalSkillName = "";
...
   {
     name: "Italian",
-    value: "it",
+    value: "it-it",
     checked: true
   },
   {
     name: "Spanish",
-    value: "es",
+    value: "es-mx",
```
Comment (Batta32) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/skill/index.js]:
```suggestion
    value: "es-es",
```

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/skill/src/copier.js [hunk 9]
```diff
@@ -6,7 +6,7 @@
 const { join } = require(`path`);
 const templateFiles = new Map();
 const deploymentTemplateFiles = new Map();
-const allLanguages = [`zh`, `de`, `en`, `fr`, `it`, `es`];
+const allLanguages = [`zh-cn`, `de-de`, `en-us`, `fr-fr`, `it-it`, `es-mx`];
```
Comment (Batta32) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/skill/src/copier.js]:
```suggestion
const allLanguages = [`zh-cn`, `de-de`, `en-us`, `fr-fr`, `it-it`, `es-es`];
```

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/skill/templates/sample-skill/deployment/scripts/deploy_cognitive_models.ps1 [hunk 10]
```diff
@@ -110,7 +110,7 @@ $settings = @{ defaultLocale = $languageArr[0]; cognitiveModels = New-Object PSO
 Write-Host "> Deploying cognitive models ..."
 foreach ($language in $languageArr)
 {
-	$langCode = ($language -split "-")[0]
+	$langCode = $language
```
Comment (Batta32) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/skill/templates/sample-skill/deployment/scripts/deploy_cognitive_models.ps1]: Is it necessary this assignment? Can we use `$language` for the rest of the `foreach`?

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/skill/templates/sample-skill/src/dialogs/_skillDialogBase.ts [hunk 11]
```diff
@@ -135,7 +135,7 @@ export class SkillDialogBase extends ComponentDialog {
             const state: SkillState = await this.stateAccessor.get(dc.context, new SkillState());
 
             // Get luis service for current locale
-            const locale: string = i18next.language.substring(0, 2);
+            const locale: string = i18next.language;
```
Comment (Batta32) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/skill/templates/sample-skill/src/dialogs/_skillDialogBase.ts]: I can't see this change in the `Sample Skill`, shoud we add it too?

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/skill/templates/sample-skill/src/index.ts [hunk 12]
```diff
@@ -49,8 +49,8 @@ import { IBotSettings } from './services/botSettings';
 // tslint:disable-next-line: no-floating-promises
 i18next.use(i18nextNodeFsBackend)
     .init({
-        fallbackLng: 'en',
-        preload: [ 'de', 'en', 'es', 'fr', 'it', 'zh' ]
+        fallbackLng: 'en-us',
+        preload: ['de-de', 'en-us', 'es-mx', 'fr-fr', 'it-it', 'zh-cn']
```
Comment (Batta32) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/skill/templates/sample-skill/src/index.ts]:
```suggestion
        preload: ['de-de', 'en-us', 'es-es', 'fr-fr', 'it-it', 'zh-cn']
```

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/skill/templates/sample-skill/src/index.ts [hunk 13]
```diff
@@ -132,7 +132,7 @@ let bot: DialogBot<Dialog>;
 try {
 
     const responseManager: ResponseManager = new ResponseManager(
-        ['en', 'de', 'es', 'fr', 'it', 'zh'],
+        ['en-us', 'de-de', 'es-mx', 'fr-fr', 'it-it', 'zh-cn'],
```
Comment (Batta32) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/skill/templates/sample-skill/src/index.ts]:
```suggestion
        ['en-us', 'de-de', 'es-es', 'fr-fr', 'it-it', 'zh-cn'],
```

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/skill/templates/sample-skill/test/helpers/skillTestBase.js [hunk 14]
```diff
@@ -71,8 +71,8 @@ const setupEnvironment = function(testMode) {
 const configuration = async function() {
   // Configure internationalization and default locale
   await i18next.use(i18nextNodeFsBackend).init({
-    fallbackLng: "en",
-    preload: ["de", "en", "es", "fr", "it", "zh"]
+    fallbackLng: 'en-us',
+    preload: ['de-de', 'en-us', 'es-mx', 'fr-fr', 'it-it', 'zh-cn'],
```
Comment (Batta32) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/skill/templates/sample-skill/test/helpers/skillTestBase.js]:
```suggestion
    preload: ['de-de', 'en-us', 'es-es', 'fr-fr', 'it-it', 'zh-cn'],
```

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/skill/templates/sample-skill/test/helpers/skillTestBase.js [hunk 15]
```diff
@@ -100,7 +100,7 @@ const initialize = async function() {
   const skillContextAccessor = userState.createProperty(SkillContext.name);
   telemetryClient = getTelemetryClient(botSettings);
   const responseManager = new ResponseManager(
-    ["en", "de", "es", "fr", "it", "zh"],
+    ['en-us', 'de-de', 'es-mx', 'fr-fr', 'it-it', 'zh-cn'],
```
Comment (Batta32) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/skill/templates/sample-skill/test/helpers/skillTestBase.js]:
```suggestion
    ['en-us', 'de-de', 'es-es', 'fr-fr', 'it-it', 'zh-cn'],
```

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/test/generator-botbuilder-assistant.skill.test.suite.js [hunk 16]
```diff
@@ -9,8 +9,8 @@ const rimraf = require(`rimraf`);
 const _kebabCase = require(`lodash/kebabCase`);
 const _camelCase = require(`lodash/camelCase`);
 const semver = require('semver');
-const languages = [`zh`, `de`, `en`, `fr`, `it`, `es`];
-const someLanguages = [`zh`, `de`, `en`];
+const languages = [`zh-cn`, `de-de`, `en-us`, `fr-fr`, `it-it`, `es-mx`];
```
Comment (Batta32) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/test/generator-botbuilder-assistant.skill.test.suite.js]:
```suggestion
const languages = [`zh-cn`, `de-de`, `en-us`, `fr-fr`, `it-it`, `es-es`];
```

File: templates/Virtual-Assistant-Template/typescript/samples/sample-assistant/deployment/scripts/deploy_cognitive_models.ps1 [hunk 17]
```diff
@@ -114,8 +114,8 @@ $settings = @{ defaultLocale = $languageArr[0]; cognitiveModels = New-Object PSO
 Write-Host "> Deploying cognitive models ..."
 foreach ($language in $languageArr)
 {
-    $langCode = ($language -split "-")[0]
 	$config = New-Object PSObject
+    $langCode = $language
```
Comment (Batta32) [inline templates/Virtual-Assistant-Template/typescript/samples/sample-assistant/deployment/scripts/deploy_cognitive_models.ps1]: Is it necessary this assignment? Can we use `$language` for the rest of the `foreach`?

File: templates/Virtual-Assistant-Template/typescript/samples/sample-skill/deployment/scripts/deploy.ps1 [hunk 18]
```diff
@@ -195,34 +195,18 @@ if ($outputs)
...
-	
-	# Publish bot
-	Write-Host "+ To publish your bot, run '$(Join-Path $PSScriptRoot 'publish.ps1') -name $($name) -resourceGroup $($resourceGroup) -projFolder `"$($projDir)` '" -ForegroundColor Magenta
-=======
 	Invoke-Expression "& '$(Join-Path $PSScriptRoot 'deploy_cognitive_models.ps1')' -name $($name) -luisAuthoringRegion $($luisAuthoringRegion) -luisAuthoringKey $($luisAuthoringKey) -luisAccountName $($outputs.luis.value.accountName) -luisAccountRegion $($outputs.luis.value.region) -luisSubscriptionKey $($outputs.luis.value.key) -resourceGroup $($resourceGroup) -qnaSubscriptionKey '$($qnaSubscriptionKey)' -outFolder '$($projDir)' -languages '$($languages)'"
 	
 	# Publish bot
 	Write-Host "+ To publish your bot, run '$(Join-Path $PSScriptRoot 'publish.ps1')' -name $($outputs.botWebAppName.value) -resourceGroup $($resourceGroup) -projFolder '$($projDir)'" -ForegroundColor Magenta
->>>>>>> c1ac5dac... Update Sample Skill
 	Write-Host "> Done."
```
Comment (Batta32) [inline templates/Virtual-Assistant-Template/typescript/samples/sample-skill/deployment/scripts/deploy.ps1]: I don't think that these changes should be in this Pull Request, you can find it in `master` branch

File: templates/Virtual-Assistant-Template/typescript/samples/sample-skill/deployment/scripts/deploy_cognitive_models.ps1 [hunk 19]
```diff
@@ -110,8 +110,8 @@ $settings = @{ defaultLocale = $languageArr[0]; cognitiveModels = New-Object PSO
 Write-Host "> Deploying cognitive models ..."
 foreach ($language in $languageArr)
 {
-	$langCode = ($language -split "-")[0]
 	$config = New-Object PSObject
+    $langCode = $language
```
Comment (Batta32) [inline templates/Virtual-Assistant-Template/typescript/samples/sample-skill/deployment/scripts/deploy_cognitive_models.ps1]: Is it necessary this assignment? Can we use `$language` for the rest of the `foreach`?

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/sample-assistant/test/helpers/testBase.js [hunk 20]
```diff
@@ -6,7 +6,7 @@
 /**
  * Possible value: record | lockdown
  */
-const TEST_MODE = 'lockdown';
+const TEST_MODE = 'record';
```
Comment (Batta32) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/sample-assistant/test/helpers/testBase.js]:
```suggestion
const TEST_MODE = 'lockdown';
```

File: templates/Virtual-Assistant-Template/typescript/samples/sample-assistant/test/helpers/testBase.js [hunk 21]
```diff
@@ -6,7 +6,7 @@
 /**
  * Possible value: record | lockdown
  */
-const TEST_MODE = 'lockdown';
+const TEST_MODE = 'record';
```
Comment (Batta32) [inline templates/Virtual-Assistant-Template/typescript/samples/sample-assistant/test/helpers/testBase.js]:
```suggestion
const TEST_MODE = 'lockdown';
```

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/sample-assistant/src/index.ts [hunk 22]
```diff
@@ -101,7 +101,7 @@ if (botSettings.cosmosDb === undefined) {
     throw new Error();
 }
 cosmosDbStorageSettings = {
-    authKey: botSettings.cosmosDb.authkey,
+    authKey: botSettings.cosmosDb.authKey,
```
Comment (DiegoCardozo94) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/sample-assistant/src/index.ts]:
```suggestion
    authKey: botSettings.cosmosDb.authkey,
```

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/sample-assistant/test/helpers/botTestBase.js [hunk 23]
```diff
@@ -80,6 +80,7 @@ async function getTestAdapterDefault(settings) {
     };
     const botServices = new BotServices(botSettings);
     const onboardingStateAccessor = userState.createProperty('OnboardingState');
+    const skillContextAccessor = userState.createProperty('skillContext');
```
Comment (DiegoCardozo94) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/sample-assistant/test/helpers/botTestBase.js]:
```suggestion
```

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/sample-assistant/test/helpers/botTestBase.js [hunk 24]
```diff
@@ -91,6 +92,7 @@ async function getTestAdapterDefault(settings) {
         escalateDialog,
         cancelDialog,
         skillDialogs,
+        skillContextAccessor,
```
Comment (DiegoCardozo94) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/app/templates/sample-assistant/test/helpers/botTestBase.js]:
```suggestion
```

File: templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/skill/templates/sample-skill/src/index.ts [hunk 25]
```diff
@@ -94,7 +94,7 @@ if (botSettings.cosmosDb === undefined) {
 }
 
 cosmosDbStorageSettings = {
-    authKey: botSettings.cosmosDb.authkey,
+    authKey: botSettings.cosmosDb.authKey,
```
Comment (DiegoCardozo94) [inline templates/Virtual-Assistant-Template/typescript/generator-botbuilder-assistant/generators/skill/templates/sample-skill/src/index.ts]:
```suggestion
    authkey: botSettings.cosmosDb.authkey,
```

File: templates/Virtual-Assistant-Template/typescript/samples/sample-assistant/src/appsettings.json [hunk 26]
```diff
@@ -9,7 +9,7 @@
     "container": ""
   },
   "cosmosDb": {
-    "authkey": "",
+    "authKey": "",
```
Comment (DiegoCardozo94) [inline templates/Virtual-Assistant-Template/typescript/samples/sample-assistant/src/appsettings.json]:
```suggestion
    "authkey": "",
```

File: templates/Virtual-Assistant-Template/typescript/samples/sample-assistant/src/index.ts [hunk 27]
```diff
@@ -101,7 +101,7 @@ if (botSettings.cosmosDb === undefined) {
     throw new Error();
 }
 cosmosDbStorageSettings = {
-    authKey: botSettings.cosmosDb.authkey,
+    authKey: botSettings.cosmosDb.authKey,
```
Comment (DiegoCardozo94) [inline templates/Virtual-Assistant-Template/typescript/samples/sample-assistant/src/index.ts]:
```suggestion
    authKey: botSettings.cosmosDb.authkey,
```

File: templates/Virtual-Assistant-Template/typescript/samples/sample-skill/src/appsettings.json [hunk 28]
```diff
@@ -9,7 +9,7 @@
     "container": ""
   },
   "cosmosDb": {
-    "authkey": "",
+    "authKey": "",
```
Comment (DiegoCardozo94) [inline templates/Virtual-Assistant-Template/typescript/samples/sample-skill/src/appsettings.json]:
```suggestion
    "authkey": "",
```

File: templates/Virtual-Assistant-Template/typescript/samples/sample-skill/src/index.ts [hunk 29]
```diff
@@ -94,7 +94,7 @@ if (botSettings.cosmosDb === undefined) {
 }
 
 cosmosDbStorageSettings = {
-    authKey: botSettings.cosmosDb.authkey,
+    authKey: botSettings.cosmosDb.authKey,
```
Comment (DiegoCardozo94) [inline templates/Virtual-Assistant-Template/typescript/samples/sample-skill/src/index.ts]:
```suggestion
    authKey: botSettings.cosmosDb.authkey,
```

**Pull request closed without merging**